### PR TITLE
Pull Request the whole module

### PR DESCRIPTION
**Pull request by apacios**

### CODE REVIEW

File: .github/workflow/php.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      max-parallel: 3
+      matrix:
+        php-versions: ['5.6', '7.3']
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+    - name: PHP syntax checker 7.3
```
Comment (zalexki) [inline .github/workflow/php.yml]: you can use `(PHP ${{ matrix.php-versions }})`

File: .github/workflow/php.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      run: find . -type f -name '*.php' ! -path "./vendor/*" ! -path "./tests/*" -exec php -l -n {} \; | (! grep -v "No syntax errors detected")
+    - name: Install Composer dependencies
+      run: composer install --no-progress --no-suggest --prefer-dist --optimize-autoloader
+    - name: Run CS Fixer
+      run: php vendor/bin/php-cs-fixer fix --dry-run --diff --using-cache=no --diff-format udiff
```
Comment (zalexki) [inline .github/workflow/php.yml]: missing empty line on EOF

File: .github/workflow/php.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    strategy:
+      fail-fast: false
+      max-parallel: 3
+      matrix:
+        php-versions: ['5.6', '7.3']
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+    - name: PHP syntax checker 7.3
+      run: find . -type f -name '*.php' ! -path "./vendor/*" ! -path "./tests/*" -exec php -l -n {} \; | (! grep -v "No syntax errors detected")
```
Comment (zalexki) [inline .github/workflow/php.yml]: just FYI we probably gonna use `https://github.com/PrestaShop/github-action-php-lint` but right now it doesnt seems to properly work so let's keep this for now

File: views/templates/admin/tabs/listing/listing_export.tpl [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+				<thead>
+					<tr>
+						<th colspan="2">{l s='Module name' mod='ps_translateyourmodule'}</th>
+						<th>{l s='Version' mod='ps_translateyourmodule'}</th>
+						<th class="table_select">{l s='See translations for' mod='ps_translateyourmodule'}</th>
+						<th colspan="2">{l s='Export translation XLSX' mod='ps_translateyourmodule'}</th>
+						<th>{l s='Export translation\'s folder' mod='ps_translateyourmodule'}</th>
+					</tr>
+				</thead>
+				<tbody>
```
Comment (zalexki) [inline views/templates/admin/tabs/listing/listing_export.tpl]: something wrong happened with indentation on this file

File: views/templates/admin/tabs/listing/import.tpl [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        <i class="material-icons">attach_file</i>{l s='Import translations' mod='ps_translateyourmodule'}
+    </div>
+	<div class="panel-body">
+    	<div class="row">
+			<div class="col-lg-2">
+				<img class="vertical-align" src="{$imagePath}international.svg"/>
+			</div>
+			<div class="col-lg-4">
+				<p>{l s='To import your translation file, the document must respect the following rules:' mod='ps_translateyourmodule'}</p>
+				<ul>
```
Comment (zalexki) [inline views/templates/admin/tabs/listing/import.tpl]: something wrong happened with indentation on this file

File: views/templates/admin/configure.tpl [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ *}
+
+<div class="bootstrap" id="ps_translateyourmodule_configuration">
+	<!-- Module content -->
+	<div id="modulecontent" class="tab-pane">
+		{include file="./tabs/listing/import.tpl"}
+		{include file="./tabs/listing/errors.tpl"}
```
Comment (zalexki) [inline views/templates/admin/configure.tpl]: something wrong happened with indentation on this file

File: views/js/admin/dropzone.min.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
...==j?1:j},h=function(a,b,c,d,e,f,h,i,j,k){var l;return l=g(b),a.drawImage(b,c,d,e,f,h,i,j,k/l)},f=function(a,b){var c,d,e,f,g,h,i,j,k;if(e=!1,k=!0,d=a.document,j=d.documentElement,c=d.addEventListener?"addEventListener":"attachEvent",i=d.addEventListener?"removeEventListener":"detachEvent",h=d.addEventListener?"":"on",f=function(c){return"readystatechange"!==c.type||"complete"===d.readyState?(("load"===c.type?a:d)[i](h+c.type,f,!1),!e&&(e=!0)?b.call(a,c.type||c):void 0):void 0},g=function(){var a;try{j.doScroll("left")}catch(b){return a=b,setTimeout(g,50),void 0}return f("poll")},"complete"!==d.readyState){if(d.createEventObject&&j.doScroll){try{k=!a.frameElement}catch(l){}k&&g()}return d[c](h+"DOMContentLoaded",f,!1),d[c](h+"readystatechange",f,!1),a[c](h+"load",f,!1)}},a._autoDiscoverFunction=function(){return a.autoDiscover?a.discover():void 0},f(window,a._autoDiscoverFunction)}).call(this)}),a.alias("component-emitter/index.js","dropzone/deps/emitter/index.js"),a.alias("component-emitter/index.js","emitter/index.js"),"object"==typeof exports?module.exports=a("dropzone"):"function"==typeof define&&define.amd?define(function(){return a("dropzone")}):this.Dropzone=a("dropzone")}();
```
Comment (zalexki) [inline views/js/admin/dropzone.min.js:1]: should be a npm dependency, not even sure if adding the license above is legal 🤔

Comment (apacios) [reply views/js/admin/dropzone.min.js:1]: Wow... The license comes from prestashop/header-stamp. Didn't see it... Thanks.
You're right concerning the npm dependency.

File: classes/File/CheckFile.php [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+     * @return bool
+     */
+    public function isAllowedTranslationFile($file, $fileName)
+    {
+        // Extension must be 'php'
+        if (self::TRANSLATION_FILE_TYPE !== substr($file, -4)) {
+            return false;
+        }
+
+        // We don't take index file
```
Comment (zalexki) [inline classes/File/CheckFile.php]: wrong comment

File: composer.json [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        "platform": {
+            "php": "5.6"
+        },
+        "optimize-autoloader": true,
+        "prepend-autoloader": false
+    },
+    "require-dev": {
+        "php": ">=5.6",
+        "phpunit/phpunit": "~5.7",
+        "guzzlehttp/guzzle": "~5.0",
```
Comment (zalexki) [inline composer.json]: guzzle has a dev dep ?

File: composer.json [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+            "PrestaShop\\Module\\PsTranslateYourModule\\": "classes/"
+        },
+        "classmap": [
+            "ps_translateyourmodule.php",
+            "controllers/"
+        ]
+    },
+    "require": {
+        "ramsey/uuid": "^3.8",
+        "vlucas/phpdotenv": "^3.4",
```
Comment (zalexki) [inline composer.json]: didn't see where you use it 🤔

File: controllers/admin/AdminDownloadZipController.php [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+*
+* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+* versions in the future. If you wish to customize PrestaShop for your
+* needs please refer to http://www.prestashop.com for more information.
+*
+* @author PrestaShop SA <contact@prestashop.com>
+* @copyright 2007-2019 PrestaShop SA
+* @license http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+* International Registered Trademark & Property of PrestaShop SA
+**/
```
Comment (zalexki) [inline controllers/admin/AdminDownloadZipController.php]: some fuckery happened here too with the header-stamp

**Review by zalexki — CHANGES_REQUESTED** [review]

**Comment by zalexki** [comment]
I'll add that the current way using regex might not be the easiest/safest.
I would try to simply boot the module and get all the translations needed with the prestashop built-in translation system that must be aware of all needed translations.

Plus i think it's @matks that told me about some kind of already built-in translation system in prestashop capable of generating needed translation keys for a module/theme (or something like that). 
So now that we have a translation module working, i think it's important that the next step is to get informations about what exists and how we can use it for our needs.

**Comment by matks** [comment]
> Plus i think it's @matks that told me about some kind of already built-in translation system in prestashop capable of generating needed translation keys for a module/theme (or something like that).
> So now that we have a translation module working, i think it's important that the next step is to get informations about what exists and how we can use it for our needs.

So, I know that:
- prestashop can translate native modules using one system A
- prestashop can translate 3rd party modules (such as this one) using one system B

And system A and system B are not the same (dont asky me why 🙄).

Also, we have enabled a system C that uses Symfony translator to be able to translate Symfony modules, and legacy modules (= modules not using Symfony) can use it too ! This one is fully documented here: https://devdocs.prestashop.com/1.7/modules/creation/module-translation/new-system/ and is backward compatible: if a module was using system B and is switching to system C, translations already created for system B can be used by system C.

However this system C is available since PS 1.7.6

File: controllers/admin/AdminAjaxPsTranslateYourModuleController.php [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+    /**
+     * Give the link to the Module's Translation page
+     * Need a module name and a lang
+     *
+     * @return void
+     */
+    public function ajaxProcessGetModuleTranslationLink()
+    {
+        $moduleName = \Tools::getValue('module', $this->module->name);
+        $lang = \Tools::getValue('lang', $this->module::DEFAULT_LANGUAGE_ISO);
```
Comment (Poulinhoo) [inline controllers/admin/AdminAjaxPsTranslateYourModuleController.php]: please change all `$this->module::` on `ps_translateyourmodule::`

File: classes/File/CheckFile.php [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+        if (self::INITIAL_LANGUAGE === $fileName) {
+            return false;
+        }
+
+        // Default language is EN, we don't need it
```
Comment (zalexki) [inline classes/File/CheckFile.php]: wrong comment

**Comment by Quetzacoalt91** [comment]
@zalexki & @matks as your comments are potential improvements, I created a dedicated issue for that. The objective in this PR is to have a first version reproducing what we currently have on the disneystore.

File: views/templates/admin/tabs/listing/listing_export.tpl (hunk 4 above)
Comment (apacios) [reply views/templates/admin/tabs/listing/listing_export.tpl]: Thanks

File: views/templates/admin/tabs/listing/import.tpl (hunk 5 above)
Comment (apacios) [reply views/templates/admin/tabs/listing/import.tpl]: Thanks

File: views/templates/admin/configure.tpl (hunk 6 above)
Comment (apacios) [reply views/templates/admin/configure.tpl]: Thanks

File: views/js/admin/dropzone.min.js (hunk 7 above)
Comment (apacios) [reply views/js/admin/dropzone.min.js:1]: Thanks

File: composer.json (hunk 9 above)
Comment (apacios) [reply composer.json]: Thanks

File: composer.json (hunk 10 above)
Comment (apacios) [reply composer.json]: Thanks

File: controllers/admin/AdminDownloadZipController.php (hunk 11 above)
Comment (apacios) [reply controllers/admin/AdminDownloadZipController.php]: Thanks

File: classes/File/CheckFile.php (hunk 8 above)
Comment (apacios) [reply classes/File/CheckFile.php]: Thanks

File: controllers/admin/AdminAjaxPsTranslateYourModuleController.php (hunk 12 above)
Comment (apacios) [reply controllers/admin/AdminAjaxPsTranslateYourModuleController.php]: Thanks

File: .github/workflow/php.yml (hunk 1 above)
Comment (apacios) [reply .github/workflow/php.yml]: Thanks

File: .github/workflow/php.yml (hunk 2 above)
Comment (apacios) [reply .github/workflow/php.yml]: Thanks

File: composer.json [hunk 14]
```diff
@@ -22,6 +22,7 @@
         ]
     },
     "require": {
-        "phpoffice/phpspreadsheet": "^1.8"
+        "phpoffice/phpspreadsheet": "^1.8",
+        "friendsofphp/php-cs-fixer": "^2.16"
```
Comment (zalexki) [inline composer.json]: we shouldnt need this as the github action only use the configuration file

Comment (apacios) [reply composer.json]: Right, thanks

**Review by zalexki — APPROVED** [review]

**Pull request merged**